neurom/core/types.py

```diff
@@ -47,25 +47,35 @@ class NeuriteIter(OrderedEnum):
     NRN = 2
 
 
-class CompositeType(int):
-    """A composite type is a list of section types.
+class Subtypes:
 
-    Each subtype corresponds to a homogeneous subtree in the morphology.
+    def __init__(self, *subtypes):
+        self.subtypes = subtypes
 
-    Args:
-        value (int): The integer value.
-        types: (List[SectionType]): The list of subtypes for given type.
-    """
+    def is_composite(self):
+        return len(self.subtypes) > 1
+
+    def __eq__(self, other):
 
-    def __new__(cls, value, types):
-        """Create an int object and assign to it the `subtypes` attribute."""
-        obj = super().__new__(cls, value)
-        obj.subtypes = types
-        return obj
+        if isinstance(self, Subtypes):
+            if isinstance(other, Subtypes):
+                if self.is_composite():
+                    if other.is_composite():
+                        return self.subtypes == other.subtypes
+                    return other.subtypes[0] in self.subtypes
+                if other.is_composite():
+                    return self.subtypes[0] in other.subtypes
+                return self.subtypes[0] == other.subtypes[0]
+            return other in self.subtypes
+
+        if isinstance(other, Subtypes):
+            return self in other.subtypes
+
+        return self == other
```

Comment:
Not super excited about this mountain range here, but neurite type is also compared with bare SectionType and sometimes with ints.